waku/v2/protocol/waku_store/protocol.nim

```diff
@@ -78,25 +74,68 @@ proc reportStoredMessagesMetric*(w: WakuStore) =
 
   waku_store_messages.set(resCount.value, labelValues = ["stored"])
 
+# TODO: Move to a message store wrapper
+proc isValidMessage(msg: WakuMessage): bool =
+  if msg.timestamp == 0:
+    return true
+
+  let 
+    now = getNanosecondTime(getTime().toUnixFloat())
+    lowerBound = now - MaxMessageTimestampVariance
+    upperBound = now + MaxMessageTimestampVariance
+
+  return lowerBound <= msg.timestamp and msg.timestamp <= upperBound
+
+# TODO: Move to a message store wrapper
+proc handleMessage*(w: WakuStore, pubsubTopic: string, msg: WakuMessage) =
```

Comment:
Should this proc return a Result as well, since there are cases of failed message handling?

Reply:
All the "message handling and storage" logic, in other words, the logic that is not related to historical message retrieval, will be extracted into a separate entity together with the changes necessary to support the "custom" message validation.

-------

That's a good question. And the answer is "It depends": for now, I would like to leave it as is since this is used in the waku relay handler (and the filter push handler), which cannot take any action if the handling is unsuccessful. They only can log the failure/error.

I also thought about that, but it will be necessary to implement the message validation functionality. Now it provides little/no value.

Reply:
Okay :)